Standards/scs-0216-v1-high-security-setup.md

```diff
@@ -0,0 +1,174 @@
+which will need to be mitigated by an administrator with the proper knowledge.
+Hardened Kubernetes clusters are desirable regardless of the possible thread model, since
+higher security doesn't necessarily mean higher complexity in this case.
+
+## Motivation
+
+Kubernetes clusters are highly configurable to fit a wide array of use-cases and setups.
+This makes them flexible and scalable to a certain extent, which is probably also why they
+found such a wide adoption. But its also obvious, that security problems can arise if a
+cluster isn't configured properly to mitigate them.
+These security risks can potentially be exposed in many different parts of a cluster, e.g.
+different APIs, authorization and authentication procedures or even Pod privilege mechanisms.
+In order to mitigate these problems, different hardening and prevention steps and mechanisms
+could be used to increase the security of a Kubernetes setup.
+
+## Design Considerations
+
+### External CA
+
+Kubernetes provides an API to provision TLS certificates that can be signed by a CA.
+This CA can be controlled by the cluster provider, which enables much more tight control
+over the clusters communication and therefore also better controllable security.
+
+In order to do this, the CA certificate bundle needs to be added to the trusted certificates
+of the server.
+To provide a certificate, the following steps need to be undertaken:
+
+1. Create a CSR
+2. Send the CSR manifest to the k8s API
+3. Approve the CSR
+4. Sign CSR with your CA
+5. Upload the signed certificate to the server
+
+This certificate could now be used by a user in a pod in order to provide a trusted certificate.
+
+It is also possible for the Kubernetes controller manager to provide the signing functionality.
+To enable this, `--cluster-signing-cert-file` and `--cluster-signing-key-file` need to be set with
+a reference to the CA keypair, which was used in the previous example to sign a CSR.
+
+### Protected Kubernetes endpoints
+
+In order to secure a Kubernetes cluster, the protection and hardening of endpoints is important.
+To do this, different approaches can be taken.
+
+#### TLS for all internal/API traffic
+
+It is already expected by Kubernetes, that all API communication internally is encrypted with TLS.
+Nevertheless, some endpoints of internal components could be/will be exposed without the necessary
+encryption, which could lead to weakpoints in the system.
+A list of the default service endpoints can be seen in the following table
+
+Protocol | Port Range | Purpose
+— | — | —
+TCP | 6443- | Kubernetes API Server
+TCP | 2379-2380 | etcd server client API
+TCP | 10250 | Kubelet API
+TCP | 10251 | kube-scheduler
+TCP | 10252 | kube-controller-manager
+TCP | 10255 | Read-Only Kubelet API
+
+Protocol | Port Range | Purpose
+— | — | —
+TCP | 10250 | Kubelet API
+TCP | 10255 | Read-Only Kubelet API
+TCP | 30000-32767 | NodePort Services
+
```

Comment:
Sidenote: The tables do not seem to render in github's MD preview.

Content: You talk about internal components being exposed without encryption and then you provide a table with all edpoints AFAICS. As a reader, I would expect you to point out which of these are unencrypted and this potentially a risk for security.

Reply:
As far as I understood from the sources I read, all of these endpoints COULD potentially be unencrypted, if the wrong configuration is done.
But I will add another column to point to some especially endagered endpoints.